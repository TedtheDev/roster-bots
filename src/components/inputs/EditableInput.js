import React, { Component } from 'react';
import styled from 'styled-components';

const Input = styled.input`
    border: none;
    outline: ${props => props.duplicateError ? `3px solid red` : `none`};
    background: ${props => props.editable ? `white` : `transparent`};
    &:focus {
        outline: 1px solid green;
    }
`;

export class EditableInput extends Component {
    constructor(props) {
        super(props);

        this.state = {editable: false}
    }

    onClickEditable = () => {
        this.setState({editable: true});
    }

    onKeyPressEnter = (event) => {
        if(event.key === 'Enter') {
            this.setState({editable: false})
        }
    }

    onBlurInput = () => {
        this.setState({editable: false});
    }

    render() {
        return (
            <Input 
                duplicateError={this.props.error}
                onClick={this.onClickEditable}
                onBlur={this.onBlurInput} 
                onKeyPress={this.onKeyPressEnter} 
                editable={this.state.editable} 
                {...this.props} 
            />
        )
    }
}

export default EditableInput;