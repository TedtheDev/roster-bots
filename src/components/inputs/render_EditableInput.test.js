import React from 'react';
import renderer from 'react-test-renderer';
import EditableInput  from './EditableInput';

describe('<EditableInput />', () => {
    it('renders correctly', () => {
        const wrapper = renderer.create(<EditableInput />);
        const tree = JSON.stringify(wrapper);

        expect(tree).toMatchSnapshot();
    });
});