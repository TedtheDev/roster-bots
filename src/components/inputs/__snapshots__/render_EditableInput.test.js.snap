// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditableInput /> renders correctly 1`] = `"{\\"type\\":\\"input\\",\\"props\\":{\\"className\\":\\"sc-bdVaJa dUqDqt\\"},\\"children\\":null}"`;
